src/mplfinance/plotting.py

```diff
@@ -909,6 +909,13 @@ def _addplot_columns(panid,panels,ydata,apdict,xdates,config):
         width  = apdict['width'] if apdict['width'] is not None else 1.6*config['_width_config']['line_width']
         alpha  = apdict['alpha']
         ax.plot(xdates,ydata,linestyle=ls,color=color,linewidth=width,alpha=alpha)
+    elif aptype == 'step':
+        where = apdict['where']
+        ls = apdict['linestyle']
+        color  = apdict['color']
+        width  = apdict['width'] if apdict['width'] is not None else 1.6*config['_width_config']['line_width']
+        alpha  = apdict['alpha']
+        ax.step(range(len(xdates)),ydata,where = where,linestyle=ls,color=color,linewidth=width,alpha=alpha)
```

Comment:
The xdata should be just *`xdates`* (*not* `range(len(xdates)`).   Then `range()` trick that I mentioned [here](https://github.com/matplotlib/mplfinance/issues/367#issuecomment-812268715) is only for the case of calling `step()` directly on the returned `axlist[]` when `show_notrading=False`.  Internally, mplfinance has ***already*** converted `xdates` [**in the code here**](https://github.com/matplotlib/mplfinance/blob/master/src/mplfinance/plotting.py#L372-L379).